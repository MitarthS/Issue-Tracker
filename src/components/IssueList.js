import React from 'react';
import {Container, Row, Col} from 'react-bootstrap';
import IssueActions from '../actions/IssueActions';
import Issue from './Issue';
export default class IssueList extends React.Component {
    // static defaultProps = { // <-- DEFAULT PROPS
    //     issues: []       // undefined gets converted to array,render won't trigger error
    // }
    constructor(){
        super();
        this.deleteIssue = this.deleteIssue.bind(this);
        this.state = {
            srch : "",
            c : 1
        };
    }
    deleteIssue(id) {
        console.log('Deleting Issue from IssueList for id .... ' + id);
        IssueActions.deleteIssue(id);
        // this.setState((prevState) => ({
        //     c: prevState.c + 1 
        // }));

    }


    // KeyPress(event){
    //     this.setState(srch = event.target.value;);
    //     console.log(`This is ${srch}`);
    // }
  

    render() {
        let issueNodes = this.props.issues.filter( (val) => {
            console.log(this.state.srch);
            if (this.state.srch === "") return val;
            else if (val.description.toLowerCase().includes( this.state.srch.toLowerCase() )) return val;
            else return null;
        } ).map(issue =>
            <Issue key = {issue.id} id = {issue.id} description={issue.description} severity={issue.severity} status={issue.status} onDelete={this.deleteIssue}>
            </Issue>
        );
        const header = {
            fontSize: '25px',
            color: 'black',
            fontWeight:'bold'
        }
        const sty = {
            width : '35%',
            marginLeft : '147px'
        }

        return (

            <Container>
                <Row>
                    <Col style={header}>Index</Col>
                    <Col xs={5} style={header}>Issue Description</Col>
                    <Col style={header}>Severity</Col>
                    <Col style={header}>Status</Col>
                    <Col></Col>
                    <Col></Col>
                </Row>
                <Row><input type="text" onChange={(e) => this.setState({srch : e.target.value})  } placeholder='Search...' style={sty}></input></Row>
                    {issueNodes}
            </Container>

        );
    }
}