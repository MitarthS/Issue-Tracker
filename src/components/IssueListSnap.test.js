import React from 'react';
import {shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import IssueList from './IssueList';

describe('When no props is passed to Issue ', () => {
    it('renders correctly', () => {
        let wrapper = shallow(<IssueList/>);
        expect(wrapper).toMatchSnapshot();
    });
});

describe('When issues array is passed to IssueList', () => {
    let wrapper;
    let props;

    beforeEach(() => {
        //srch = "";
        props = {
            issues: [
                {
                    "id":1,
                    "description": "Raj",
                    "serverity": "Critical",
                    "status": "Open"                    
                },
                {
                    "id":2,
                    "description": "Ramesh",
                    "serverity": "Critical",
                    "status": "Open"                    
                }

            ]
        }
        //val = "";
        wrapper = shallow(<IssueList {...props}/>);
    });

    it('renders correctly when passed the  props', () => {
        expect(wrapper).toMatchSnapshot();
    });

});

describe('When issues empty array passed to IssueList', () => {
    let wrapper;
    let props;

    beforeEach(() => {
        props = {
            issues: []
        }
        wrapper = shallow(<IssueList {...props}/>);
    });

    it('should not crash', () => {
        expect(wrapper).toMatchSnapshot();
    });

});
