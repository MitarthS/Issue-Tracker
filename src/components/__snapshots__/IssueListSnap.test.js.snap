// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When issues array is passed to IssueList renders correctly when passed the  props 1`] = `ShallowWrapper {}`;

exports[`When issues empty array passed to IssueList should not crash 1`] = `ShallowWrapper {}`;

exports[`When no props is passed to Issue  renders correctly 1`] = `ShallowWrapper {}`;
